lib/client.js

```diff
@@ -201,10 +201,6 @@ class Client extends EventEmitter {
       throw new InvalidArgumentError('invalid url')
     }
 
```

Comment:
I think I would prefer something like:

```js
if (url.port == null) {
  if (protocol === 'http:') {
    url = { ...url, port: 80 }
  } else if (protocol === 'https:') {
    url = { ...url, port: 443 }
  }
}
```

Reply:
That would work only if the url would be a plain object, but if we use the `URL` constructor (as we are, ad we should probably enforce), restructuring it will result in a not really nice object:
```
> var url = new URL('http://localhost')
undefined
> url
URL {
  href: 'http://localhost/',
  origin: 'http://localhost',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'localhost',
  hostname: 'localhost',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> { ...url, port: 80 }
{
  port: 80,
  [Symbol(context)]: URLContext {
    flags: 400,
    scheme: 'http:',
    username: '',
    password: '',
    host: 'localhost',
    port: null,
    path: [ '' ],
    query: null,
    fragment: null
  },
  [Symbol(query)]: URLSearchParams {}
}
```

Reply:
Maybe we should add a new validation like the following:
```js
if (typeof url !== 'string' && url instanceof URL !== true) {
  throw new InvalidArgumentError('invalid url')
}
```

Reply:
No, we should allow url:like objects.

Reply:
Out of curiosity, why? URL is the whatwg standard, and given that it has some subtle behavior, like the handling of `80` and `443`, supporting both the URL and a url-like object looks like a bug-prone thing to support.

Furthermore, but this is my personal opinion, configuration is the root of all bugs :D

Reply:
I have use cases where we have an options object with `port, hostname, path` it seems silly and error prone to have to do:

`${protocol}//${hostname}${port ? `:${port}`: ''}${path}`

Reply:
I personally don't like the `URL` way because a valid URL can contain stuff that we don't want, e.g. path and well strings are unstructured.